lib/SyTest/Federation/Room.pm

```diff
@@ -0,0 +1,138 @@
+package SyTest::Federation::Room;
+
+use strict;
+use warnings;
+
+sub make_event_refs
+{
+   [ map { [ $_->{event_id}, $_->{hashes} ] } @_ ];
+}
+
+sub create
+{
+   my $class = shift;
+   my %args = @_;
+
+   my $server = $args{server};
+
+   my $creator = $args{creator};
+   my $room_id = $args{room_id} // $server->next_room_id;
+
+   my $self = bless {
+      room_id => $room_id,
+      server  => $server,
+
+      current_state => {},
+      prev_events => [],
+   }, $class;
+
+   $self->create_event(
+      type => "m.room.create",
+
+      content     => { creator => $creator },
+      depth       => 0,
+      sender      => $creator,
+      state_key   => "",
+   );
+
+   $self->create_event(
+      type => "m.room.member",
+
+      content     => { membership => "join" },
+      depth       => 0,
```

Comment:
I would naively assume that the join and join_rules would be deeper than the creation. Is this not the case?